packages/amplify-graphql-name-mapping-transformer/src/__tests__/graphql-refers-to-transformer.test.ts

```diff
@@ -0,0 +1,166 @@
+    stubTransformerContext.modelToDatasourceMap.set('DDBModel', { dbType: 'DDB', provisionDB: false });
+    stubDirective.arguments = [];
+    expect(() =>
+      refersToTransformer.object(stubDefinition as ObjectTypeDefinitionNode, stubDirective as DirectiveNode, stubTransformerContext),
+    ).toThrowErrorMatchingInlineSnapshot(`"@refersTo is not supported on type DDBModel. It can only be used on a @model type."`);
+  });
+
+  it('requires a name to be specified', () => {
+    const [stubDefinition, stubDirective, stubTransformerContext] = getTransformerInputsFromSchema(simpleSchema, modelName);
+    stubDirective.arguments = [];
+    expect(() =>
+      refersToTransformer.object(stubDefinition as ObjectTypeDefinitionNode, stubDirective as DirectiveNode, stubTransformerContext),
+    ).toThrowErrorMatchingInlineSnapshot(`"name is required in @refersTo directive."`);
+  });
+
+  it('requires a string value for name', () => {
+    const [stubDefinition, stubDirective, stubTransformerContext] = getTransformerInputsFromSchema(simpleSchema, modelName);
+    stubDirective.arguments![0].value.kind = 'ListValue';
+    expect(() =>
+      refersToTransformer.object(stubDefinition as ObjectTypeDefinitionNode, stubDirective as DirectiveNode, stubTransformerContext),
+    ).toThrowErrorMatchingInlineSnapshot(`"A single string must be provided for \\"name\\" in @refersTo directive"`);
+  });
+
+  it('registers the rename mapping', () => {
+    const [stubDefinition, stubDirective, stubTransformerContext] = getTransformerInputsFromSchema(simpleSchema, modelName);
+    refersToTransformer.object(stubDefinition as ObjectTypeDefinitionNode, stubDirective as DirectiveNode, stubTransformerContext);
+    expect(setModelNameMapping_mock.mock.calls[0]).toEqual(['TestName', 'OriginalName']);
+  });
+
+  it('throws if a conflicting model name is present in the schema', () => {
+    const [stubDefinition, stubDirective, stubTransformerContext] = getTransformerInputsFromSchema(conflictingModelSchema, modelName);
+    expect(() =>
+      refersToTransformer.object(stubDefinition as ObjectTypeDefinitionNode, stubDirective as DirectiveNode, stubTransformerContext),
+    ).toThrowErrorMatchingInlineSnapshot(
+      `"Cannot apply @refersTo with name \\"OriginalName\\" on type \\"TestName\\" because \\"OriginalName\\" model already exists in the schema."`,
+    );
+  });
+
+  it('throws if a model name mapping is duplicate', () => {
+    const [stubDefinition, stubDirective, stubTransformerContext] = getTransformerInputsFromSchema(
+      duplicateNameMappingModelSchema,
+      modelName,
+    );
+    expect(() =>
+      refersToTransformer.object(stubDefinition as ObjectTypeDefinitionNode, stubDirective as DirectiveNode, stubTransformerContext),
+    ).toThrowErrorMatchingInlineSnapshot(
+      `"Cannot apply @refersTo with name \\"OriginalName\\" on type \\"TestName\\" because \\"DuplicateName\\" model already has the same name mapping."`,
```

Comment:
Do we throw an error if customer is using refersTo on DDB models?

Reply:
Yes we do, the test pointed out below i.e `can be applied only on RDS models` does that check. Updated the below test.